docs/vision-usage.rst

```diff
@@ -0,0 +1,190 @@
+Using the Vision API
+====================
+
+Authentication and Configuration
+--------------------------------
+
+- For an overview of authentication in ``gcloud-python``,
+  see :doc:`gcloud-auth`.
+
+- In addition to any authentication configuration, you should also set the
+  :envvar:`GCLOUD_PROJECT` environment variable for the project you'd like
+  to interact with. If you are Google App Engine or Google Compute Engine
+  this will be detected automatically.
+
+- After configuring your environment, create a
+  :class:`Client <gcloud.vision.client.Client>`
+
+  .. doctest::
+
+     >>> from gcloud import vision
+     >>> client = vision.Client()
+
+  or pass in ``credentials`` and ``project`` explicitly
+
+  .. doctest::
+
+     >>> from gcloud import vision
+     >>> client = vision.Client(project='my-project', credentials=creds)
+
+Annotating an Image
+-------------------
+
+Annotate a single image
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. doctest::
+
+    >>> from gcloud import vision
+    >>> client = vision.Client(project="my-project-name")
+    >>> with open('/tmp/car.jpg', 'r') as f:
+    ...     client.annotate(f.read(), vision.FeatureTypes.LABEL_DETECTION, 3)
+
+Annotate multiple images
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. doctest::
+
+     >>> images = (
+     ...     ('./image.jpg', [
+     ...         vision.FeatureTypes.LABEL_DETECTION,
+     ...         vision.FeatureTypes.LANDMARK_DETECTION]),
+     ...     ('./image2.jpg', [
+     ...         vision.FeatureTypes.FACE_DETECTION,
+     ...         vision.FeatureTypes.TEXT_DETECTION]),)
+     >>> annotated_images = []
+     >>> for image, feature_types in images:
+     ...     annotated_images.append(
+     ...         vision_client.annotate(
```

Comment:
See my note above... IMO, we should let people write code the straight-forward way, tell us what items should go into the same batch (ie, you don't need a until you're out of the context), and we do the magic under the hood. So the example above is the "ideal case", and we throw exceptions if you try to use something that we don't have yet...? ie

```python
image1 = client.image('/path/to/image1.jpg')

with client.batch():
  faces = image1.detect_faces(limit=10)
  if faces:  # raise ValueError on __iter__ or __bool__ here
             # saying you're asking for something in the middle of a batch.
    labels = image1.detect_labels()
```